subprojects/persistent-cache/src/main/java/org/gradle/cache/Cache.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2011 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.cache;
+
+import javax.annotation.Nullable;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public interface Cache<K, V>  {
+    /**
+     * Locates the given entry, using the supplied factory when the entry is not present or has been discarded, to recreate the entry in the cache.
+     *
+     * <p>Implementations may prevent more than one thread calculating the same key at the same time or not.
+     */
+    V get(K key, Function<? super K, ? extends V> factory);
+
+    default V get(K key, Supplier<? extends V> supplier) {
+        return get(key, __ -> supplier.get());
+    }
```

Comment:
Did you consider adding a variant with `V defaultValue` as well?

Reply:
I did not. Do we use that variant anywhere?

I considered renaming the method to `computeIfAbsent()`, though I think `get` is better since we are talking about a cache. Maybe `load` would be even better as a method name?

Reply:
`getOrLoad()`?

Reply:
FWIW, Guava and Caffeine declare `Cache.get(K, Function)`, so we could stay with `get`.

Reply:
You might consider using `get` / `getIfPresent` to call out the active vs passive methods, like we did.

Map is passive data structure by default, so it calls out `computeIfPresent` to highlight the side effects. A cache is active, so side effects happen by default (e.g. resetting expiration timestamps), so we call out passive methods where side effects are reduced. Neither overloads `get` to mean both compute or ifPresent, and I think it's worth naming those differently.

Reply:
I used `getIfPresent` in https://github.com/gradle/gradle/pull/15067.